include/zephyr/drivers/flash.h

```diff
@@ -255,7 +255,27 @@ struct flash_pages_info {
 	uint32_t index;
 };
 
+struct flash_erase_region {
+	off_t start_offset;	/* offset from the base address of flash */
+	size_t size;		/* remaining size from start_offset */
+	size_t erase_block_size;
```

Comment:
I think we should have some naming scheme in the structure similar to the Flash Area where every member starts with prefix like `fer_`. We could probably shorten `start_offset` to offset.

Reply:
I will reduce `start_offset` to `offset`.

Naming, a very difficult subject. IMO the Flash Area naming is not the way to follow, it is confusing and makes reading/writing code difficult e.g. offset is referred to as:
* for a flash area variable `fa`:`fa.fa_off`,
* for a flash area variable `area`: `area.fa_off`,
what is the value of adding the abbreviated type ?

Reply:
Adding a bit to this, Could we use size_t for the offset? I don't see any reason to mix signed and unsigned values when the offset can not be negative :)

Also, I would not prepend `fer_`, since the member is already name spaced by the struct (as @Laczen also points out)

Reply:
It is easier to find code that messes with fields of a structure. I guess that `start_offset` or `size` may get quite common in other structures. On the other hand it is quite easy to figure out that anything that touches `fa_off`  has to deal with `flash_area` object.

The prefixing is useful when you start to embed object one into another. You will immediately figure out that you are writing to the wrong object when `fa_off` assignment generates error that structure does not have such member. Unless of course you have two object and mix them, but that is another thing.

Reply:
I don't completely understand this, if you embed object into another the embedded object also gets a name:
```
struct high_level {
        ...
        const struct flash_area fa; /* or (*fa) */
};

struct high_level ha = {
        ...
        fa.fa_off = .,
}
```